src/map/controls/queryby.js

```diff
@@ -227,36 +208,33 @@ export class QueryBy extends InteractionControl {
                     map.showAddLayerModal();
                   }
 
-                  if (!['__ALL__', '__NEW__'].includes(value) && value !== getSelectedLayerId() ) {
+                  const selected = (GUI.getService('map').getSelectedLayer() || { getId() { return '__ALL__'; } }).getId(); // TODO: use optional chaining instead: GUI.getService('map').getSelectedLayer()?.getId() || '__ALL__'
+
+                  if (!['__ALL__', '__NEW__'].includes(value) && value !== selected) {
                     map.selectLayer(value);
                   }
-                  //reset selection if a selection is done by TOC catalog
-                  if (['__ALL__', '__NEW__'].includes(value) && '__ALL__' !== getSelectedLayerId()) {
+
+                  // reset selection when done through TOC catalog
+                  if (['__ALL__', '__NEW__'].includes(value) && '__ALL__' !== selected) {
```

Comment:
@volterra79 double check.

Duplicated condition on `__ALL__`, superfluous?

```js
const selected = (GUI.getService('map').getSelectedLayer() || { getId() { return '__ALL__'; } }).getId()

...

if (['__ALL__', '__NEW__'].includes(value) && '__ALL__' !== selected) {
...
```

Reply:
misread, not duplicates

Reply:
@Raruto https://github.com/g3w-suite/g3w-client/pull/762/commits/06a244d5d5ab24723ea04e5ac9d946843dddd0c9